pipeline/DESIGN.md

```diff
@@ -0,0 +1,212 @@
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Type of the stream, see [stream type values](#stream-types)                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | Type of the event, see [event type values](#event-types)                                        |
+| state       | bytes             | JSON encoding of the event state                                                                |
+
+### Transformation
+
+Computes the singular head that is the canonical state of the stream.
+
+
+## Enumeration Values
+
+### Stream Types
+
+| name                    | code | description                                                | specification                                                                                       |
+| ----                    | ---- | -----------                                                | -------------                                                                                       |
+| Tile                    | 0x00 | A stream type representing a json document                 | https://cips.ceramic.network/CIPs/cip-8                                                             |
+| CAIP-10 Link            | 0x01 | Link blockchain accounts to DIDs                           | https://cips.ceramic.network/CIPs/cip-7                                                             |
+| Model                   | 0x02 | Defines a schema shared by group of documents in ComposeDB | https://github.com/ceramicnetwork/js-ceramic/tree/main/packages/stream-model                        |
+| Model Instance Document | 0x03 | Represents a json document in ComposeDB                    | https://github.com/ceramicnetwork/js-ceramic/tree/main/packages/stream-model-instance               |
+| UNLOADABLE              | 0x04 | A stream that is not meant to be loaded                    | https://github.com/ceramicnetwork/js-ceramic/blob/main/packages/stream-model/src/model.ts#L163-L165 |
+| EventId                 | 0x05 | An event id encoded as a cip-124 EventID                   | https://cips.ceramic.network/CIPs/cip-124                                                           |
```

Comment:
I forgot we did this.  Do you remember why we need this?

Reply:
> encoded as a cip-124 EventID

Except in this context isn't it encoded as a StreamID?

Reply:
We should also probably clarify in the description that this isn't a "real" StreamType.  Similar to UNLOADABLE

Reply:
Not sure, just copied from the spec. I'll add a note.